### PR TITLE
Get visdiff running again

### PR DESCRIPTION
**Pull request by chromakode**
It appears that we've got a heisenbug where rarely the electron process never starts up. I haven't been able to figure this out, so I've added some more debug output and a timeout. Also, now visdiff failures won't break your build, though they will send me a slack notification. (If you'd like to be included in the slack notification, lmk!) This way we can get visdiff working 95% of the time, and continue trying to figure out the issues when the hanging behavior rears its head again.

:eyeglasses: @keybase/react-hackers

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...6513724849a1 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474071947359-b62d7693fb42-6513724849a1-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474071947359-b62d7693fb42-6513724849a1-linux/New_Proof_PGP_generating-Generating_PGP-6513724849a1.png)  
   New_Proof_PGP_generating-Generating_PGP comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...c1f0e8a1fcb0 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474308947725-b62d7693fb42-c1f0e8a1fcb0-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474308947725-b62d7693fb42-c1f0e8a1fcb0-linux/New_Proof_PGP_generating-Generating_PGP-c1f0e8a1fcb0.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474308947725-b62d7693fb42-c1f0e8a1fcb0-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474308947725-b62d7693fb42-c1f0e8a1fcb0-linux/Icon-Icon_Image-c1f0e8a1fcb0.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...e42703e111c2 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474314215652-b62d7693fb42-e42703e111c2-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474314215652-b62d7693fb42-e42703e111c2-linux/New_Proof_PGP_generating-Generating_PGP-e42703e111c2.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474314215652-b62d7693fb42-e42703e111c2-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474314215652-b62d7693fb42-e42703e111c2-linux/Icon-Icon_Image-e42703e111c2.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...4cff421e32bf introduced some visual changes on linux:
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474314744244-b62d7693fb42-4cff421e32bf-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474314744244-b62d7693fb42-4cff421e32bf-linux/Icon-Icon_Image-4cff421e32bf.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...1bbb09d8f0dc introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474315458683-b62d7693fb42-1bbb09d8f0dc-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474315458683-b62d7693fb42-1bbb09d8f0dc-linux/New_Proof_PGP_generating-Generating_PGP-1bbb09d8f0dc.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474315458683-b62d7693fb42-1bbb09d8f0dc-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474315458683-b62d7693fb42-1bbb09d8f0dc-linux/Icon-Icon_Image-1bbb09d8f0dc.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...e394a20d3429 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474316184686-b62d7693fb42-e394a20d3429-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474316184686-b62d7693fb42-e394a20d3429-linux/New_Proof_PGP_generating-Generating_PGP-e394a20d3429.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474316184686-b62d7693fb42-e394a20d3429-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474316184686-b62d7693fb42-e394a20d3429-linux/Icon-Icon_Image-e394a20d3429.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...66f0ebe8c9b3 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474316268577-b62d7693fb42-66f0ebe8c9b3-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474316268577-b62d7693fb42-66f0ebe8c9b3-linux/New_Proof_PGP_generating-Generating_PGP-66f0ebe8c9b3.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474316268577-b62d7693fb42-66f0ebe8c9b3-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474316268577-b62d7693fb42-66f0ebe8c9b3-linux/Icon-Icon_Image-66f0ebe8c9b3.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...09c070847769 introduced some visual changes on linux:
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474317333505-b62d7693fb42-09c070847769-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474317333505-b62d7693fb42-09c070847769-linux/Icon-Icon_Image-09c070847769.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...c9f335fd2c24 introduced some visual changes on linux:
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474317713875-b62d7693fb42-c9f335fd2c24-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474317713875-b62d7693fb42-c9f335fd2c24-linux/Icon-Icon_Image-c9f335fd2c24.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...6df5f80dd07a introduced some visual changes on linux:
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474318258217-b62d7693fb42-6df5f80dd07a-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474318258217-b62d7693fb42-6df5f80dd07a-linux/Icon-Icon_Image-6df5f80dd07a.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...b669cd2b8002 introduced some visual changes on linux:
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474318962555-b62d7693fb42-b669cd2b8002-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474318962555-b62d7693fb42-b669cd2b8002-linux/Icon-Icon_Image-b669cd2b8002.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits b62d7693fb42...534a3b365ab4 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474319615434-b62d7693fb42-534a3b365ab4-linux/New_Proof_PGP_generating-Generating_PGP-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474319615434-b62d7693fb42-534a3b365ab4-linux/New_Proof_PGP_generating-Generating_PGP-534a3b365ab4.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474319615434-b62d7693fb42-534a3b365ab4-linux/Icon-Icon_Image-b62d7693fb42.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474319615434-b62d7693fb42-534a3b365ab4-linux/Icon-Icon_Image-534a3b365ab4.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 33810aca4381...1785c2480900 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474330186047-33810aca4381-1785c2480900-linux/New_Proof_PGP_generating-Generating_PGP-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474330186047-33810aca4381-1785c2480900-linux/New_Proof_PGP_generating-Generating_PGP-1785c2480900.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474330186047-33810aca4381-1785c2480900-linux/Icon-Icon_Image-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474330186047-33810aca4381-1785c2480900-linux/Icon-Icon_Image-1785c2480900.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 33810aca4381...5797e12463ff introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474330399971-33810aca4381-5797e12463ff-linux/New_Proof_PGP_generating-Generating_PGP-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474330399971-33810aca4381-5797e12463ff-linux/New_Proof_PGP_generating-Generating_PGP-5797e12463ff.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474330399971-33810aca4381-5797e12463ff-linux/Icon-Icon_Image-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474330399971-33810aca4381-5797e12463ff-linux/Icon-Icon_Image-5797e12463ff.png)  
   Icon-Icon_Image comparison

File: Jenkinsfile [hunk 1]
```diff
@@ -138,29 +138,34 @@ if (env.CHANGE_TITLE && env.CHANGE_TITLE.contains('[ci-skip]')) {
...
+                                                }
+                                                sh "npm install ./visdiff"
+                                                try {
+                                                    timeout(time: 10, unit: 'MINUTES') {
+                                                        dir("desktop") {
+                                                            sh "../node_modules/.bin/keybase-visdiff 'merge-base(origin/master, HEAD)...HEAD'"
+                                                        }
+                                                    }
+                                                } catch (e) {
+                                                    helpers.slackMessage("#ci-notify", "warning", "<@mgood>: visdiff failed: <${env.BUILD_URL}|${env.JOB_NAME} ${env.BUILD_DISPLAY_NAME}>")
```
Comment (chrisnojima) [inline Jenkinsfile]: maybe we should post to its own channel so we can join to come in / out vs having to edit the code?

File: desktop/test/render-visdiff.js [hunk 2]
```diff
@@ -61,15 +64,16 @@ app.on('ready', () => {
...
   })
 
-  ipcMain.on('visdiff-ready', ev => renderNext(ev.sender))
-
   const scriptPath = resolveRoot('dist', 'visdiff.bundle.js')
   for (let i = 0; i < WORKER_COUNT; i++) {
     setTimeout(() => {
       const workerWin = new BrowserWindow({show: false, width: CANVAS_SIZE, height: CANVAS_SIZE})
+      console.log('Created new worker window', i)
       // TODO: once we're on electron v1.2.3, try ready-to-show event.
```
Comment (chrisnojima) [inline desktop/test/render-visdiff.js]: dunno if we want to at this moment but we are using a newer electron than 1.2.3 so maybe we could change this?

File: Jenkinsfile (hunk 1 above)
Comment (chromakode) [inline Jenkinsfile]: Good idea! Done.

File: desktop/test/render-visdiff.js (hunk 2 above)
Comment (chromakode) [inline desktop/test/render-visdiff.js]: Oh yeah! I'll take another look, thanks for pointing this out.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 33810aca4381...ed536cebb153 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474395728160-33810aca4381-ed536cebb153-linux/New_Proof_PGP_generating-Generating_PGP-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474395728160-33810aca4381-ed536cebb153-linux/New_Proof_PGP_generating-Generating_PGP-ed536cebb153.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474395728160-33810aca4381-ed536cebb153-linux/Icon-Icon_Image-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474395728160-33810aca4381-ed536cebb153-linux/Icon-Icon_Image-ed536cebb153.png)  
   Icon-Icon_Image comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 33810aca4381...ddd427fa075f introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474396171162-33810aca4381-ddd427fa075f-linux/New_Proof_PGP_generating-Generating_PGP-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474396171162-33810aca4381-ddd427fa075f-linux/New_Proof_PGP_generating-Generating_PGP-ddd427fa075f.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474396171162-33810aca4381-ddd427fa075f-linux/Icon-Icon_Image-33810aca4381.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474396171162-33810aca4381-ddd427fa075f-linux/Icon-Icon_Image-ddd427fa075f.png)  
   Icon-Icon_Image comparison

**Pull request merged**